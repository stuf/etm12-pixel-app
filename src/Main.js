/* eslint no-unused-vars: [1, {"varsIgnorePattern": "[K|T|Model]"}] */
/**
 * @module Main
 */
import * as React from 'karet';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import SplashScene from 'scenes/Splash';
import EditorScene from 'scenes/Editor';
import LoadImageScene from 'scenes/LoadImage';
import NotFoundScene from 'scenes/NotFound';
import TestPanelsScene from 'scenes/test/TestPanels';

function MainScene({ routerProps = {}, ...props }) {
  const WithRootProps = (Component, extra = {}) => p => (
    <Component {...{ ...props, ...p, ...extra }} />
  );

  return (
    <main className="layout--main debugRoot">
      <Router {...routerProps}>
        <Switch>
          <Route path="/load" component={WithRootProps(LoadImageScene)} />
          <Route path="/editor" component={WithRootProps(EditorScene)} />
          <Route
            exact
            path="/"
            component={WithRootProps(SplashScene, { redirectTo: '/editor' })}
          />
          <Route path="/test">
            <Switch>
              <Route
                path="/test/splash"
                component={WithRootProps(SplashScene)}
              />
              <Route path="/test/panels" component={TestPanelsScene} />
            </Switch>
          </Route>
          <Route component={WithRootProps(NotFoundScene)} />
        </Switch>
      </Router>
    </main>
  );
}

export default MainScene;
