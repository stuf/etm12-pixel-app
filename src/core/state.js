/* eslint no-unused-vars: [1, {"varsIgnorePattern": "[K|T]"}] */
/**
 * @module state
 * @namespace core
 */
import * as U from 'karet.util';

import palettes from './palettes';

import * as S from 'settings';
import * as T from './state.d';

/**
 * @type {T.IState}
 */
const initialState = {
  canvas: {
    size: [32, 32],
    scale: 16,
  },
  currentFile: {
    name: 'Untitled',
    createdAt: new Date(),
    isEditing: false,
  },
  color: {
    currentColor: 0,
    currentPalette: 0,
    palettes,
  },
  app: {
    devMode: S.app.devMode,
  },
  devtool: {
    flags: {
      offsetGuide: true,
      showBuildInfo: true,
    },
  },
};

const state = U.atom(initialState);

export default state;
