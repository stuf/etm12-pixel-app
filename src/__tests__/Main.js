import * as React from 'karet';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import state from 'core/state';
import canvasData from 'core/canvas-data';

import Main from '../Main';

it('matches snapshot', () => {
  const wrap = shallow(<Main />);
  expect(toJson(wrap)).toMatchSnapshot();
});

it('works', () => {
  const wrapper = mount(<Main />);
});
