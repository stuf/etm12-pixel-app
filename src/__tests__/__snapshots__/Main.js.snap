// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<main
  className="layout--main"
>
  <BrowserRouter>
    <Switch>
      <Route
        component={[Function]}
        path="/editor"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        path="/test"
      >
        <Switch>
          <Route
            component={[Function]}
            path="/test/splash"
          />
          <Route
            path="/test/panels"
          />
        </Switch>
      </Route>
      <Route
        component={[Function]}
        exact={true}
        path="/test/panels"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </BrowserRouter>
</main>
`;
